### PR TITLE
MM-46456 mattermost airflow operator

### PR DESCRIPTION
**Pull request by munish7771**
This PR is result of code changes w.r.t mattermost-operator from previous PR -> https://github.com/mattermost/mattermost-data-warehouse/pull/1029

Reference -> https://github.com/ifoukarakis/airflow-provider-mattermost

#### Summary

- [x] Created MattermostWebhookHook to handle http requests instead of python requests module.
- [x] Updated the MattermostOperator to use MattermostWebhookHook.
- [x] Added a sample dag to test operator run.
- [x] Added airflow-requirements.txt
- [x] Added tests for operator and hook.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-46456

### CODE REVIEW

File: plugins/hooks/mattermost_webhook_hook.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from airflow.hooks.http_hook import HttpHook
+
+
+class MattermostWebhookHook(HttpHook):
+
+    """
+    Send messages to Mattermost using webhooks. For more details on the parameters, please check
+    `Incoming webhooks <https://developers.mattermost.com/integrate/webhooks/incoming/#parameters>_` documentation.
+
+    :param http_conn_id: :ref:`http connection<howto/connection:http>` that has the base
```
Comment (ifoukarakis) [inline plugins/hooks/mattermost_webhook_hook.py:14]:
```suggestion
    :param mattermost_conn_id: :ref:`http connection<howto/connection:http>` that has the base
```

Small inconsistency between docs and code.

File: plugins/operators/mattermost_operator.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+from airflow.plugins_manager import AirflowPlugin
+# airflow.providers.http.operators.http -> airflow 2
```
Comment (ifoukarakis) [inline plugins/operators/mattermost_operator.py:2]:
```suggestion
```

Let's remove commented out code.

File: plugins/operators/mattermost_operator.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from airflow.operators.http_operator import SimpleHttpOperator
+from hooks.mattermost_webhook_hook import *
+
+
+class MattermostOperator(SimpleHttpOperator):
+
+    """
+    Operator that allows sending messages to Mattermost. For more details on the parameters, please check
+    `Incoming webhooks <https://developers.mattermost.com/integrate/webhooks/incoming/#parameters>_` documentation.
+    :param http_conn_id: :ref:`http connection<howto/connection:http>` that has the base
```
Comment (ifoukarakis) [inline plugins/operators/mattermost_operator.py:12]:
```suggestion
    :param mattermost_conn_id: :ref:`http connection<howto/connection:http>` that has the base
```

Small inconsistency between docs and code.

File: dags/development/mattermost_operator.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from airflow import DAG
```
Comment (ifoukarakis) [inline dags/development/mattermost_operator.py:1]: This looks like a test dag. Should it be deployed @ production?

**Review by ifoukarakis — COMMENTED** [review]
🚀 Looks awesome! Just left a couple of small notes/questions.

File: plugins/hooks/mattermost_webhook_hook.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    def execute(self) -> None:
+        msg = self._build_message()
+        self.run(
+            endpoint=self._build_endpoint(),
+            data=msg,
+            headers={'Content-type': 'application/json'},
+            extra_options={'check_response': True},
+        )
+
+class MattermostPlugin(AirflowPlugin):
```
Comment (catalintomai) [inline plugins/hooks/mattermost_webhook_hook.py:81]: Where is this being used?

Comment (munish7771) [reply plugins/hooks/mattermost_webhook_hook.py:80]: in case theresponse status code is other than 200, it will raise an exception.

**Review by ifoukarakis — APPROVED** [review]

**Pull request merged**